tests/e2e/setup.go

```diff
@@ -74,103 +86,122 @@ func NewCCVTestSuite[Tp e2eutil.ProviderApp, Tc e2eutil.ConsumerApp](
 type SetupCallback func(t *testing.T) (
 	coord *ibctesting.Coordinator,
 	providerChain *ibctesting.TestChain,
-	consumerChain *ibctesting.TestChain,
 	providerApp e2eutil.ProviderApp,
-	consumerApp e2eutil.ConsumerApp,
+	consumerBundles map[string]*icstestingutils.ConsumerBundle,
 )
 
 // SetupTest sets up in-mem state before every test
 func (suite *CCVTestSuite) SetupTest() {
 
 	// Instantiate new test utils using callback
 	suite.coordinator, suite.providerChain,
-		suite.consumerChain, suite.providerApp,
-		suite.consumerApp = suite.setupCallback(suite.T())
-
-	providerKeeper := suite.providerApp.GetProviderKeeper()
-	consumerKeeper := suite.consumerApp.GetConsumerKeeper()
-
-	// valsets must match
-	providerValUpdates := tmtypes.TM2PB.ValidatorUpdates(suite.providerChain.Vals)
-	consumerValUpdates := tmtypes.TM2PB.ValidatorUpdates(suite.consumerChain.Vals)
-	suite.Require().True(len(providerValUpdates) == len(consumerValUpdates), "initial valset not matching")
-	for i := 0; i < len(providerValUpdates); i++ {
-		addr1 := utils.GetChangePubKeyAddress(providerValUpdates[i])
-		addr2 := utils.GetChangePubKeyAddress(consumerValUpdates[i])
-		suite.Require().True(bytes.Equal(addr1, addr2), "validator mismatch")
+		suite.providerApp, suite.consumerBundles = suite.setupCallback(suite.T())
```

Comment:
If done correctly, none of the following code should change how setup is done. It should merely add for-loops to setup all consumers in the exact same way we previously setup a single consumer